support-lib/jni/Marshal.hpp

```diff
@@ -607,12 +612,22 @@ namespace djinni
         {
             // Serialize to C++ vector
             std::vector<uint8_t> cppBuf(c.ByteSizeLong());
```

Comment:
The size is `c.ByteSizeLong()`, so once we know the size is 0, there is no need to construct the `vector<>` any more.

Reply:
The vector lifetime needs to extend beyond the size check to the `deserializeJavaProto()` call